tests/functional/func_test.py

```diff
@@ -89,6 +89,8 @@ def run_testcase(test_name, config, verbose, test_case):
     if not stop_syslogng():
       sys.exit(1)
     print_end(test_name, success)
+    if not success:
+        failed_tests.append(test_name)
```

Comment:
This is useful, thanks.

Optional (just an opinion): I'd probably append the `failed_tests` global outside to avoid using more globals in these functions.